### ISSUE TITLE
New rule: `lines-around-comment`

### ISSUE THREAD

**Issue by hansbergren** [typescript-eslint/typescript-eslint#1933]
I'd like to use the [lines-around-comment](https://eslint.org/docs/rules/lines-around-comment) rule from ESLint, however, the rule does not appear to parse TypeScript interfaces as blocks, since it warns on the following code:

```ts
export interface AcceptMediaType {
  // The raw Accept type
  mediaType: string
}
```

even though I allow comments to start a block:
```js
{
    'lines-around-comment': [
      'warn',
      {
        beforeBlockComment: true,
        afterBlockComment: false,
        beforeLineComment: true,
        afterLineComment: false,
        allowBlockStart: true,
        allowBlockEnd: true,
        allowObjectStart: true,
        allowObjectEnd: true,
        allowArrayStart: true,
        allowArrayEnd: true,
        allowClassStart: true,
        allowClassEnd: true,
      },
    ],
}
```

I think having a type-aware version of this rule would be beneficial.

**Comment by dkamyshov** [comment]
Looks like a duplicate of #1150
